charts/xp-management/templates/deployment.yaml

```diff
@@ -4,6 +4,9 @@ metadata:
   name: {{ include "management-svc.fullname" . }}
   labels:
     {{- include "management-svc.labels" . | nindent 4 }}
+    {{- with .Values.deployment.extraLabels }}
```

Comment:
1. Support for `.Values.extraLables` already exists, though it's not documented in the values file. We are relying on this in our internal deployment as well. Can it be re-used?
2. If we are to change this lables field, I think it should be under `deployment` as done in this PR. But, Merlin and Turing are calling it just `labels` as opposed to `extraLabels`. What would be the preferred convention? Should we try to make them all the same?

Reply:
Agree with that, This can be just `labels` under `deployment`.

I intend to add support for deployment to have extra labels than the chart added common labels for every resource.

Since Turing and Merlin have it as `.Values.deployment.labels`, I think we can make this also the same. wdyt?

Reply:
> I intend to add support for deployment to have extra labels than the chart added common labels for every resource.

Not sure if I understood correctly. If the goal is just to make the deployment resource created have extra labels, that behavior is already supported ([this](https://github.com/caraml-dev/helm-charts/blob/main/charts/xp-management/templates/deployment.yaml#L6) includes the extraLabels).

> Since Turing and Merlin have it as `.Values.deployment.labels`, I think we can make this also the same. wdyt?

Sounds good to me.

Btw I just realised Merlin has a separate [podLabels](https://github.com/caraml-dev/helm-charts/blob/main/charts/merlin/values.yaml#L13) too. And MLP has a [podLabels](https://github.com/caraml-dev/helm-charts/blob/main/charts/mlp/values.yaml#L14) as well as an [extraLabels](https://github.com/caraml-dev/helm-charts/blob/main/charts/mlp/values.yaml#L41). 😓 It seems this is done to differentiate between pod vs deployment labels. Anyway, perhaps unifying everything is beyond the scope of this PR and can be dealt with separately later.

Reply:
> Not sure if I understood correctly. If the goal is just to make the deployment resource created have extra labels, that behavior is already supported ([this](https://github.com/caraml-dev/helm-charts/blob/main/charts/xp-management/templates/deployment.yaml#L6) includes the extraLabels).

Yeah, This will do the job, But this will also add the extra labels to all the other resources where `"management-svc.labels"` helper is used. 

>Anyway, perhaps unifying everything is beyond the scope of this PR and can be dealt with separately later.

Will take up a task later to standardise labels across charts under caraml.